### PR TITLE
Add i2p

### PR DESCRIPTION
**Pull request by prayank23**
+ Rename 'Tor' section to 'Privacy'
+ Add 'i2p' chapter in 'Privacy' section
+ Change chapter numbers accordingly

I have added basic steps to run Bitcoin Core with i2p router. No changes in scripts because i2p support was recently [added in Bitcoin Core](https://github.com/bitcoin/bitcoin/pull/20685), less people using it, there can be bugs and it can be considered experimental at this moment. 

i2p documentation will be added in Bitcoin Core once https://github.com/bitcoin/bitcoin/pull/22250 is merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,14 +86,16 @@ _This tutorial assumes that you have some minimal background of how to use the c
...
-* [14.0: Using Tor](14_0_Using_Tor.md)
-  * [14.1: Verifying Your Tor Setup](14_1_Verifying_Your_Tor_Setup.md)
-  * [14.2: Changing Your Bitcoin Hidden Services](14_2_Changing_Your_Bitcoin_Hidden_Services.md)
-  * [14.3: Adding SSH Hidden Services](14_3_Adding_SSH_Hidden_Services.md)
+* [14.1: Using Tor](14_1_Using_Tor.md)
+  * [14.1.1: Verifying Your Tor Setup](14_1_1_Verifying_Your_Tor_Setup.md)
+  * [14.1.2: Changing Your Bitcoin Hidden Services](14_1_2_Changing_Your_Bitcoin_Hidden_Services.md)
+  * [14.1.3: Adding SSH Hidden Services](14_1_3_Adding_SSH_Hidden_Services.md)
+
+* [14.2: Using i2p](14_2_Using_i2p.md)
```
Comment (shannona) [inline README.md:99]: We don't have any other chapter split up in this manner, so lets stick with the standard. 

Leave 14.1 to 14.3 as they are.

Create a new chapter 15 header (15.0) and call it something like "Using Other Privacy Tools". It should generally match the style of 14.0, 13.0, and the other master chapters. It may be a lot briefer because we just have this one tool at this point. Maybe more in the future?

then, lay in your new section as 15.1.

Yeah, this is going to create a duplicate chapter 15. We'll get everything else renumbered at some point in the future.

File: 14_2_Using_i2p.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Chapter 14.2: Using i2p
+
+The Invisible Internet Project (I2P) is a fully encrypted private network layer. It uses a distributed [network database](https://geti2p.net/en/docs/how/network-database) and encrypted unidirectional tunnels between your and your peers.
+
+Basic differences between Tor and i2p: https://geti2p.net/en/comparison/tor
```
Comment (shannona) [inline 14_2_Using_i2p.md:5]: We'd prefer to see a list of the most important differences (in your own words), and then link to that for more.

File: 14_2_Using_i2p.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Trade-offs: https://bitcoin.stackexchange.com/questions/107060/tor-and-i2p-tradeoffs-in-bitcoin-core
+
```
Comment (shannona) [inline 14_2_Using_i2p.md:8]: Ditto for the Trade-offs. 

Just bullet point the biggest few things, and let people click through for the rest.

File: 14_2_Using_i2p.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+It is not installed by [Bitcoin Standup](https://github.com/BlockchainCommons/Bitcoin-Standup-Scripts) right now as i2p support was recently added in Bitcoin core. However, you can try it manually by following the below steps:
+
+1. Install `i2pd`: https://i2pd.readthedocs.io/en/latest/user-guide/install/
+2. Run `i2pd`: [`sudo systemctl start i2pd.service`](https://i2pd.readthedocs.io/en/latest/user-guide/run/)
+3. Confirm if `i2pd` is running: `ss -nlt`
+
```
Comment (shannona) [inline 14_2_Using_i2p.md:14]: For commands, let's stick with the of showing the command-line options.

So, you'd have something like this:

You can install `i2pd` with the [installation instructions](URL).

Then, run it and check that it's running:
```
# sudo systemctl start i2pd.service
# ss -nlt
```

File: 14_2_Using_i2p.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+   ```
+   State   Recv-Q   Send-Q     Local Address:Port      Peer Address:Port  Process
+
+   LISTEN  0        4096           127.0.0.1:7656           0.0.0.0:*
+   ```
```
Comment (shannona) [inline 14_2_Using_i2p.md:19]: I assume this is i2pd? you'll want to say something like, "You should see it running on port 7656".

File: 14_2_Using_i2p.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+4. `bitcoin.conf`:
+
```
Comment (shannona) [inline 14_2_Using_i2p.md:21]: We generally want to use imperatives to tell the reader what to do. For example, "Add the following to your bitcoin.conf file, usually found in ~/.bitcoin/bitcoin.conf" (I think!)

File: 14_2_Using_i2p.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+   ```
+   testnet=1
```
Comment (shannona) [inline 14_2_Using_i2p.md:23]: The testnet flag should already be there or not, not something to add now.

File: 14_2_Using_i2p.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+   i2psam=127.0.0.1:7656
+   debug=i2p
+   ```
+5. Run `bitcoind`
+
```
Comment (shannona) [inline 14_2_Using_i2p.md:28]: This is going to be restart bitcoind at this point, so we should show the command.

File: 14_2_Using_i2p.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+6. Check `debug.log`:
+   ```
+   2021-06-15T20:36:16Z i2paccept thread start
+   2021-06-15T20:36:16Z I2P: Creating SAM session with 127.0.0.1:7656
+
+   2021-06-15T20:36:56Z I2P: SAM session created: session id=3e0f35228b, my address=bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p:18333
+   2021-06-15T20:36:56Z AddLocal(bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p:18333,4)
+   ```
```
Comment (shannona) [inline 14_2_Using_i2p.md:36]: Check it for what? We want to let people know what they want to see (and maybe what they don't).

File: 14_2_Using_i2p.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+   anchors.dat  chainstate         i2p_private_key  settings.json
+   banlist.dat  debug.log          mempool.dat      wallets
+   blocks       fee_estimates.dat  peers.dat
+   ```
+8. `bitcoin-cli -netinfo` or `bitcoin-cli getnetworkinfo`returns i2p address:
+
+   ```
+   Local addresses
+   bmwyyuzyqdc5dcx27s4baltbu6zw7rbqfl2nmclt45i7ng3ul4pa.b32.i2p     port  18333    score      4
+   ```
```
Comment (shannona) [inline 14_2_Using_i2p.md:50]: It might be this is still in process? But I feel like we need to see more about how to use i2p at this point, etc. (Or if *nothing* more is required, what it'll do and how it'll be useful.)

And then of course the standard conclusion that ends out each section.

**Review by shannona — COMMENTED** [review]
Great to see this work come in!

I've got basically two comments for developing it:

(1) More need to go with the standard formats and styles we've developed throughout the document.
(2) Questions which I thought would be answered.

Thanks! Looking forward to seeing the next revision!

File: README.md (hunk 1 above)
Comment (prayank23) [reply README.md:99]: > Create a new chapter 15 header (15.0) and call it something like "Using Other Privacy Tools". It should generally match the style of 14.0, 13.0, and the other master chapters.

"Using i2p" will be better same as "Using Tor" because we are explaining similar thing here (creating hidden service for Bitcoin Core). I will have to add 1 to everything below [this](https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line#part-five-programming-with-rpc)

>  It may be a lot briefer because we just have this one tool at this point. Maybe more in the future?

Yes

> then, lay in your new section as 15.1

Okay. Will do this today.

> Yeah, this is going to create a duplicate chapter 15. We'll get everything else renumbered at some point in the future.

Or we can renumber in the same PR. Will be easy and its not code that will change any functionality for users, just the numbers in doc.

File: 14_2_Using_i2p.md (hunk 2 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:5]: Still researching about differences and discussing with few people. I agree it will be better if I mention few differences and provide links for details.

File: 14_2_Using_i2p.md (hunk 3 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:8]: > Just bullet point the biggest few things, and let people click through for the rest.

Sure. Waiting for someone to respond on SE today. If not, will add my own answer and continue as you suggested.

File: 14_2_Using_i2p.md (hunk 5 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:19]: > I assume this is i2pd? 

Yes

> you'll want to say something like, "You should see it running on port 7656"

Sure. Will add it.

File: 14_2_Using_i2p.md (hunk 6 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:21]: Makes sense. Will add it.

File: 14_2_Using_i2p.md (hunk 7 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:23]: Will remove it.

File: 14_2_Using_i2p.md (hunk 9 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:36]: Maybe rephrase the things mentioned here: https://github.com/vasild/bitcoin/blob/i2p_doc/doc/i2p.md#i2p-related-information-in-bitcoin-core

File: 14_2_Using_i2p.md (hunk 10 above)
Comment (prayank23) [reply 14_2_Using_i2p.md:50]: > It might be this is still in process? But I feel like we need to see more about how to use i2p at this point, etc. (Or if nothing more is required, what it'll do and how it'll be useful.)

Agree. Want to start with basic things and do follow up PRs to add more information as I learn more about i2p.

> And then of course the standard conclusion that ends out each section.

Okay

File: README.md (hunk 1 above)
Comment (shannona) [reply README.md:99]: Sounds great.

**Comment by shannona** [comment]
Overall, sounding good. Just let me know when you're ready for additional review & merge.

**Comment by prayank23** [comment]
> Overall, sounding good. Just let me know when you're ready for additional review & merge.

Made changes in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 9 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Rephrased in last commit https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 7 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Removed in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 6 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Done in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 5 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Done in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 4 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Made changes in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: 14_2_Using_i2p.md (hunk 3 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Nobody answered on SE. I have added an answer. Also mentioned major trade-offs in bullet points.

File: 14_2_Using_i2p.md (hunk 2 above)
Comment (prayank23) [reply 14_2_Using_i2p.md]: Done. Added a table in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

File: README.md (hunk 1 above)
Comment (prayank23) [reply README.md:100]: Made changes in https://github.com/BlockchainCommons/Learning-Bitcoin-from-the-Command-Line/commit/e6faf1cc07f4477c6508bbf82b1dc59940f101d9

**Comment by shannona** [comment]
This looks good. I'm going to do a minor edit of it in the near future, and may have some questions, but I think it's ready to merge as a first draft. Thanks!

**Review by shannona — APPROVED** [review]
Overall, looks good, much clearer now and with strong explanation of what we're doing. Thanks!

**Comment by shannona** [comment]
Can you resolve the conflicts with 17_2_Accessing_Bitcoind_with_Java.md? I believe some newer text is now on the repo which needs to be reflected.

**Comment by shannona** [comment]
Thank you! I had to do some additional work from the command line, but I think it's all properly integrated now
